packages/x-tree-view/src/TreeItem2/TreeItem2.tsx

```diff
@@ -113,13 +115,22 @@ export const TreeItem2Label = styled('div', {
   name: 'MuiTreeItem2',
   slot: 'Label',
   overridesResolver: (props, styles) => styles.label,
-})(({ theme }) => ({
+  shouldForwardProp: (prop) => shouldForwardProp(prop) && prop !== 'editable',
+})<{ editable?: boolean }>(({ theme }) => ({
   width: '100%',
   boxSizing: 'border-box', // prevent width + padding to overflow
   // fixes overflow - see https://github.com/mui/material-ui/issues/27372
   minWidth: 0,
   position: 'relative',
   ...theme.typography.body1,
+  variants: [
+    {
+      props: ({ editable }: UseTreeItem2LabelSlotOwnProps) => editable === true,
```

Comment:
@flaviendelangle What do you think about adding this prop to the label? 🤔 I think I could also fix the horizontal layout shift with a negative margin on the `inputLabel` and avoid any style changes on the label, but that could potentially lead to weird overlaps in custom layouts I think...

Reply:
I fixed the `editable===true` in the meantime 🤦‍♀️